cwms-data-api/src/main/java/cwms/cda/api/watersupply/AccountingCatalogController.java

```diff
@@ -0,0 +1,202 @@
+                    || Boolean.parseBoolean(ctx.queryParam(START_TIME_INCLUSIVE));
+            final boolean endInclusive = ctx.queryParam(END_TIME_INCLUSIVE) == null
+                    || Boolean.parseBoolean(ctx.queryParam(END_TIME_INCLUSIVE));
+            final boolean ascending = ctx.queryParam(ASCENDING) == null
+                    || Boolean.parseBoolean(ctx.queryParam(ASCENDING));
+            final int rowLimit = ctx.queryParam(ROW_LIMIT) != null ? Integer.parseInt(ctx.queryParam(ROW_LIMIT)) : 0;
+            DSLContext dsl = getDslContext(ctx);
+
+            String formatHeader = ctx.header(Header.ACCEPT) != null ? ctx.header(Header.ACCEPT) : Formats.JSONV1;
+            ContentType contentType = Formats.parseHeader(formatHeader, WaterSupplyAccounting.class);
+            ctx.contentType(contentType.toString());
+            CwmsId projectLocation = new CwmsId.Builder().withOfficeId(office).withName(locationId).build();
+
+            WaterContractDao contractDao = new WaterContractDao(dsl);
+            WaterUser waterUser = contractDao.getWaterUser(projectLocation, waterUserName);
+            List<WaterUserContract> contract = contractDao.getAllWaterContracts(projectLocation,
+                    waterUser.getEntityName());
+
+            if (waterUser.getEntityName() == null) {
+                CdaError error = new CdaError("Unable to retrieve accounting - no water user found for the"
+                        + " provided parameters.");
+                LOGGER.log(Level.SEVERE, "Error retrieving water pump accounting - no water user found.");
+                ctx.status(HttpServletResponse.SC_NOT_FOUND).json(error);
+                return;
+            }
+
+            boolean contractExists = false;
+            for (WaterUserContract contractItem : contract) {
+                if (contractItem.getContractId().getName().equals(contractId)) {
+                    contractExists = true;
+                    break;
+                }
+            }
+
+            if (!contractExists) {
+                CdaError error = new CdaError("Unable to retrieve accounting - no matching contract found for the"
+                        + " provided parameters.");
+                LOGGER.log(Level.SEVERE, "Error retrieving water pump accounting - no contract found.");
+                ctx.status(HttpServletResponse.SC_NOT_FOUND).json(error);
+                return;
+            }
+
+            Instant startInstant = DateUtils.parseUserDate(startTime, "UTC").toInstant();
+            Instant endInstant = DateUtils.parseUserDate(endTime, "UTC").toInstant();
+
+            WaterSupplyAccountingDao waterSupplyAccountingDao = getWaterSupplyAccountingDao(dsl);
+            List<WaterSupplyAccounting> accounting = waterSupplyAccountingDao.retrieveAccounting(contractId, waterUser,
+                    projectLocation, null, startInstant, endInstant, startInclusive, endInclusive,
```

Comment:
What is the null parameter here? Since this is the only usage of the method, there shouldn't be nulls